### DISCUSSION TITLE
ROCm 4.2 - pp_od_clk_voltage file doesn't exist anymore

### DISCUSSION THREAD

**Discussion by valeriob01** [preda/gpuowl#226] [General]
A new way to undervolt gpus is needed.

**Comment by dgcampea** [comment]
`rocm-smi --setvc` can be used to undervolt. It should also be a bit more robust in general for GPU tuning than messing with /sysfs.

**Reply by valeriob01** [reply]
> `rocm-smi --setvc` can be used to undervolt. It should also be a bit more robust in general for GPU tuning than messing with /sysfs.

Have you actually tested it?

```
# rocm-smi --showvc


======================= ROCm System Management Interface =======================
============================= Voltage Curve Points =============================
ERROR: 2 GPU[0]: od volt: RSMI_STATUS_NOT_SUPPORTED: This function is not supported in the current environment.	
GPU[0]		: Voltage Curve is not supported
================================================================================
WARNING:  		 One or more commands failed
============================= End of ROCm SMI Log ==============================

```

**Reply by dgcampea** [reply]
Works for me™
```
$ rocm-smi --showvc
========================ROCm System Management Interface========================
GPU[0] 		: Voltage point 0: 808Mhz 725mV
GPU[0] 		: Voltage point 1: 1304Mhz 801mV
GPU[0] 		: Voltage point 2: 1801Mhz 1084mV
==============================End of ROCm SMI Log ==============================
```

What kernel and ROCm installation do you have?

**Reply by valeriob01** [reply]
> Works for me™
> 
> ```
> $ rocm-smi --showvc
> ========================ROCm System Management Interface========================
> GPU[0] 		: Voltage point 0: 808Mhz 725mV
> GPU[0] 		: Voltage point 1: 1304Mhz 801mV
> GPU[0] 		: Voltage point 2: 1801Mhz 1084mV
> ==============================End of ROCm SMI Log ==============================
> ```
> 
> What kernel and ROCm installation do you have?

kernel 5.4.0-73 and ROCm 4.2.0

**Reply by dgcampea** [reply]
ROCm 4.2.0 requires installing the rock-dkms package. You also might need to add `amdgpu.ppfeaturemask=0xfffd7fff` kernel option. (check kernel docs for more info about the values)
Personally I've noticed that starting from 4.0.0, a reboot seems required for ROCm to work every time it gets updated.

**Reply by valeriob01** [reply]
> ROCm 4.2.0 requires installing the rock-dkms package. You also might need to add `amdgpu.ppfeaturemask=0xfffd7fff` kernel option. (check kernel docs for more info about the values)
> Personally I've noticed that starting from 4.0.0, a reboot seems required for ROCm to work every time it gets updated.

I have rock-dkms installed (version 4.2-21) and have set up grub commandline options to amdgpu.ppfeaturemask=0xfffd7fff , also tried with amdgpu.ppfeaturemask=0xffffffff , both times I got same result as before.

**Reply by dgcampea** [reply]
Skimming through ROCm docs it doesn't seem to mention any changes to rocm-smi, my guess is that its system config related.

- Is secureboot enabled?
- Is the rock-dkms module being loaded?
- Does ROCm 4.2 work for gpuowl?
- Can you list the contents of the directory where `pp_od_clk_voltage` is at?
- What Distro are you using?

**Reply by valeriob01** [reply]
> Skimming through ROCm docs it doesn't seem to mention any changes to rocm-smi, my guess is that its system config related.
> 
>     * Is secureboot enabled?
> 
>     * Is the rock-dkms module being loaded?
> 
>     * Does ROCm 4.2 work for gpuowl?
> 
>     * Can you list the contents of the directory where `pp_od_clk_voltage` is at?
> 
>     * What Distro are you using?

- Secureboot not enabled, and never have used it.
- rock-dkms not being loaded
- ROCm 4.2 working with gpuowl every day 24/7
- ls /sys/class/drm/card1/device/
```
# ls /sys/class/drm/card1/device/
aer_dev_correctable       msi_irqs
aer_dev_fatal             numa_node
aer_dev_nonfatal          pcie_bw
ari_enabled               pcie_replay_count
boot_vga                  power
broken_parity_status      power_dpm_force_performance_level
class                     power_dpm_state
config                    pp_cur_state
consistent_dma_mask_bits  pp_dpm_dcefclk
current_link_speed        pp_dpm_fclk
current_link_width        pp_dpm_mclk
d3cold_allowed            pp_dpm_pcie
device                    pp_dpm_sclk
df_cntr_avail             pp_dpm_socclk
dma_mask_bits             pp_features
driver                    pp_force_state
driver_override           pp_mclk_od
drm                       pp_num_states
enable                    pp_power_profile_mode
fw_version                pp_sclk_od
gpu_busy_percent          pp_table
gpu_metrics               product_name
hwmon                     product_number
i2c-10                    ras
i2c-11                    remove
i2c-5                     rescan
i2c-6                     reset
i2c-7                     resource
i2c-8                     resource0
i2c-9                     resource0_wc
irq                       resource2
link                      resource2_wc
local_cpulist             resource4
local_cpus                resource5
max_link_speed            revision
max_link_width            rom
mem_busy_percent          serial_number
mem_info_gtt_total        subsystem
mem_info_gtt_used         subsystem_device
mem_info_vis_vram_total   subsystem_vendor
mem_info_vis_vram_used    thermal_throttling_logging
mem_info_vram_total       uevent
mem_info_vram_used        unique_id
mem_info_vram_vendor      vbios_version
modalias                  vendor
msi_bus

```
- Ubuntu 20.04

**Reply by dgcampea** [reply]
> I have rock-dkms installed (version 4.2-21) and have set up grub commandline options to amdgpu.ppfeaturemask=0xfffd7fff , also tried with amdgpu.ppfeaturemask=0xffffffff , both times I got same result as before.

Just to be sure, did you regenerate grub.cfg after editing the `/etc/default/grub`? (`sudo update-grub`)

**Reply by valeriob01** [reply]
> > I have rock-dkms installed (version 4.2-21) and have set up grub commandline options to amdgpu.ppfeaturemask=0xfffd7fff , also tried with amdgpu.ppfeaturemask=0xffffffff , both times I got same result as before.
> 
> Just to be sure, did you regenerate grub.cfg after editing the `/etc/default/grub`? (`sudo update-grub`)

To be absolutely sure I just redid update-grub ... same result as before.

**Comment by selroc** [comment]
I see by reading the forum that other have stumbled in this issue. I repeat that Radeon Pro VII can't be overclocked/undervolted due to restrictions.
